Imagine/Cache/Resolver/PsrCacheResolver.php

```diff
@@ -0,0 +1,267 @@
+<?php
+
+/*
+ * This file is part of the `liip/LiipImagineBundle` project.
+ *
+ * (c) https://github.com/liip/LiipImagineBundle/graphs/contributors
+ *
+ * For the full copyright and license information, please view the LICENSE.md
+ * file that was distributed with this source code.
+ */
+
+namespace Liip\ImagineBundle\Imagine\Cache\Resolver;
+
+use Liip\ImagineBundle\Binary\BinaryInterface;
+use Psr\Cache\CacheItemInterface;
+use Psr\Cache\CacheItemPoolInterface;
+use function str_replace;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+class PsrCacheResolver implements ResolverInterface
```

Comment:
could we make this class final and use private instead of protected on the fields and methods? would make live easier if we have to change anything on the class later, when we can be sure it is not extended by people.

if somebody needs a customer resolver, they can implement the interface.